sdk/src/main/scala/com/horizen/storage/SidechainStateStorage.scala

```diff
@@ -158,14 +172,19 @@ class SidechainStateStorage(storage: Storage, sidechainBoxesCompanion: Sidechain
       new ByteArrayWrapper(WithdrawalEpochInfoSerializer.toBytes(withdrawalEpochInfo))))
 
     if (withdrawalRequestAppendSeq.nonEmpty) {
-      val withdrawalRequestList = getWithdrawalRequests(withdrawalEpochInfo.epoch)
-
-      withdrawalRequestList.addAll(withdrawalRequestAppendSeq.asJava)
+      val withdrawalRequestSeq = getWithdrawalRequests(withdrawalEpochInfo.epoch) ++ withdrawalRequestAppendSeq
```

Comment:
Instead of duplicating the list of widthrawal requests at each update, and moreover keeping it forever, we have to keep just the list of withdrawal requests that are related to the current epoch and the previous one. To implement this in an efficient way we are going to store the widthdrawals requests individually using as key the epochNumber+widthrawalReqEpochCnt. Moreover to be able to calculate faster the current epoch withdrawal request counter during the append operations we will also keep track of the current epoch counter as a separate entry (just one for all the epcohs) currentEpochWidthrawalReqCnt: counter.
As an additional optimization we are going to remove all the entries of epoch number - 2 at the beginning of each epoch.